crates/wasmparser/src/binary_reader.rs

```diff
@@ -101,7 +101,7 @@ impl BinaryReaderError {
 }
 
 /// A binary reader of the WebAssembly structures and types.
-#[derive(Clone, Debug, Hash)]
+#[derive(Clone, Debug, Hash, Eq, PartialEq)]
```

Comment:
I do not need `PartialEq` for this, it was a transitive dependency of `ConstExpr`. Considering `ConstExpr` has only one field which is a `BinaryReader`, what would be the best way to derive `Eq` and `PartialEq` for that?

Reply:
Could the traits manually be implemented for `ConstExpr`? Ideally that would actually compare opcodes themselves rather than the binary encoding of opcodes, but it's ok if it just checks the binary encodings are the same for now.

Reply:
How would you manually implement one for `ConstExpr` considering that the only field of that struct is a `BinaryReader`? How can you get access to the binary encodings?

Reply:
I believe there's a `remaining_bytes()` method (or similar) on `BinaryReader` to get access to the raw bytes.

Reply:
Done!